src/evalfunc.c

```diff
@@ -5517,16 +5522,35 @@ f_getregion(typval_T *argvars, typval_T *rettv)
 	return;
     p2 = *fp;
 
-    pos1 = tv_get_string(&argvars[0]);
-    pos2 = tv_get_string(&argvars[1]);
-    type = tv_get_string(&argvars[2]);
+    if (argvars[0].v_type == VAR_STRING)
+	pos1 = tv_get_string(&argvars[0]);
+    if (argvars[1].v_type == VAR_STRING)
+	pos2 = tv_get_string(&argvars[1]);
 
     is_visual = (pos1[0] == 'v' && pos1[1] == NUL)
 	|| (pos2[0] == 'v' && pos2[1] == NUL);
 
     if (is_visual && !VIsual_active)
 	return;
 
+    default_type[0] = VIsual_mode;
+    default_type[1] = NUL;
```

Comment:
I don't think it's a good idea to use `VIsual_mode` as the default type, as it seems that changing the motion force of an operator can also change `VIsual_mode`. I think it's better to have a deterministic default value "v".